autoload/tagbar/types/uctags.vim

```diff
@@ -560,6 +560,16 @@ function! tagbar#types#uctags#init(supported_types) abort
         \ 'struct' : 's'
     \ }
     let types.go = type_go
+    " Haskell {{{1
+    let type_haskell = tagbar#prototypes#typeinfo#new()
+    let type_haskell.ctagstype = 'haskell'
+    let type_haskell.kinds = [
+        \ {'short' : 'm', 'long' : 'module',         'fold' : 0, 'stl' : 0},
+        \ {'short' : 't', 'long' : 'types',          'fold' : 0, 'stl' : 0},
+        \ {'short' : 'c', 'long' : 'constructors',   'fold' : 0, 'stl' : 0},
+        \ {'short' : 'f', 'long' : 'functions',      'fold' : 0, 'stl' : 0}
```

Comment:
Makes sense, thanks, tried it. It seems ctors / types don't make sense to scope, because types are typically one-liners anyway, and ctors are either one-liners (for sum types), or multi-liner for record-types, but then the record accessors are treated as functions by ctags, so they get the focus anyway.